.github/workflows/test.yml

```
@@ -17,23 +17,23 @@ jobs:
         fetch-depth: 1
     - uses: actions/setup-go@v5
       with:
-        go-version: 1.18.x
+        go-version: 1.21.x
     - name: golangci-lint
       uses: golangci/golangci-lint-action@v6.1.1
       with:
-        version: v1.45.0
+        version: v1.51.2
         args: --verbose
 
   test-unit:
     strategy:
       matrix:
-        os: [ ubuntu-20.04, windows-2022 ]
+        os: [ ubuntu-20.04, macos-14, windows-2022 ]
```

```suggestion
        os: [ ubuntu-20.04, macOS-15, windows-2022 ]
```

[macos-15](https://github.com/actions/runner-images/blob/main/images/macos/macos-15-Readme.md) is available